### PR TITLE
Bug fix for activity reporting

### PR DESCRIPTION
**Pull request by philipabbey**
Added 'has' clauses around additional ActivityMonitor.Info fields that are not present on all devices. I have also checked the other reporting fields to ensure they are not device dependent.

### CODE REVIEW

**Comment by philipabbey** [comment]
@KPWhiver I'm inviting you to the team, when you've accepted the invitation I'll add you as a review for this issue.

File: source/WebhookManager.mc [hunk 1]
```diff
@@ -224,59 +225,64 @@ class WebhookManager {
...
+                "state"               => System.getSystemStats().charging,
+                "type"                => "binary_sensor",
+                "unique_id"           => "battery_is_charging",
+                "icon"                => System.getSystemStats().charging ? "mdi:battery-plus" : "mdi:battery-minus",
+                "entity_category"     => "diagnostic",
+                "disabled"            => !Settings.isSensorsLevelEnabled()
             },
             {
-                "name"        => "Steps today",
-                "state"       => activityInfo.steps == null ? "unknown" : activityInfo.steps,
```
Comment (KPWhiver) [inline source/WebhookManager.mc:238]: "unknown" is a (semi)official Home Assistant state, indicating that the sensor does not have a known value. As such, I would not replace this with a translated string.

File: source/WebhookManager.mc [hunk 2]
```diff
@@ -224,59 +225,64 @@ class WebhookManager {
...
+                "state"               => activityInfo.steps == null ? unavailable : activityInfo.steps,
+                "type"                => "sensor",
+                "unique_id"           => "steps_today",
+                "icon"                => "mdi:walk",
+                "state_class"         => "total",
+                "disabled"            => !Settings.isSensorsLevelEnabled()
             },
             {
                 "name"                => "Heart rate",
-                "state"               => heartRate == null ? "unknown" : heartRate,
```
Comment (KPWhiver) [inline source/WebhookManager.mc:247]: Some comment about "unknown" as above.

File: source/WebhookManager.mc [hunk 3]
```diff
@@ -224,59 +225,64 @@ class WebhookManager {
...
-                "icon"        => "mdi:stairs-down",
-                "state_class" => "total",
-                "disabled"    => !Settings.isSensorsLevelEnabled()
             }
         ];
 
+        if (ActivityMonitor.Info has :floorsClimbed) {
+            sensors.add({
+                "name"                => "Floors climbed today",
+                "state"               => activityInfo.floorsClimbed == null ? unavailable : activityInfo.floorsClimbed,
```
Comment (KPWhiver) [inline source/WebhookManager.mc:261]: Same comment about "unknown" as above.

File: source/WebhookManager.mc [hunk 4]
```diff
@@ -224,59 +225,64 @@ class WebhookManager {
...
+                "icon"                => "mdi:stairs-up",
+                "state_class"         => "total",
+                "disabled"            => !Settings.isSensorsLevelEnabled()
+            });
+        }
+
+        if (ActivityMonitor.Info has :floorsDescended) {
+            sensors.add({
+                "name"                => "Floors descended today",
+                "state"               => activityInfo.floorsDescended == null ? unavailable : activityInfo.floorsDescended,
```
Comment (KPWhiver) [inline source/WebhookManager.mc:273]: Same comment about "unknown" as above.

File: source/WebhookManager.mc [hunk 5]
```diff
@@ -224,59 +225,64 @@ class WebhookManager {
...
+                "state_class"         => "total",
+                "disabled"            => !Settings.isSensorsLevelEnabled()
+            });
+        }
 
         if (ActivityMonitor.Info has :respirationRate) {
             sensors.add({
                 "name"                => "Respiration rate",
-                "state"               => activityInfo.respirationRate == null ? "unknown" : activityInfo.respirationRate,
+                "state"               => activityInfo.respirationRate == null ? unavailable : activityInfo.respirationRate,
```
Comment (KPWhiver) [inline source/WebhookManager.mc:285]: Same comment about "unknown" as above.

**Review by KPWhiver — COMMENTED** [review]
Looks fine, except for the "unknown" change.

File: source/WebhookManager.mc (hunk 1 above)
Comment (philipabbey) [reply source/WebhookManager.mc:238]: Done.

File: source/WebhookManager.mc (hunk 2 above)
Comment (philipabbey) [reply source/WebhookManager.mc]: Done.

File: source/WebhookManager.mc (hunk 3 above)
Comment (philipabbey) [reply source/WebhookManager.mc]: Done.

File: source/WebhookManager.mc (hunk 4 above)
Comment (philipabbey) [reply source/WebhookManager.mc]: Done.

File: source/WebhookManager.mc (hunk 5 above)
Comment (philipabbey) [reply source/WebhookManager.mc]: Done.

**Comment by KPWhiver** [comment]
Just now realise, this also requires has clauses in the background delegate.

**Review by KPWhiver — CHANGES_REQUESTED** [review]
To fix this bug, you'll also have to add has clauses to the background delegate.

**Comment by philipabbey** [comment]
> To fix this bug, you'll also have to add has clauses to the background delegate.

Goes to show how flakey testing can be on these development tools. I have now invoked the background service from the simulation menu.

**Review by KPWhiver — APPROVED** [review]

**Comment by philipabbey** [comment]
@KPWhiver Thank you for your assistance with this amendment.

**Pull request merged**